docs/guides/03-web-components.md

```diff
@@ -0,0 +1,94 @@
+## Web components
+
+Airship provides a set of [web components](https://www.webcomponents.org/introduction). One of the key features of the Web Components standard is the ability to offer custom elements that encapsulate functionality on an HTML page, rather than having to make do with a long, nested batch of elements that together provide a custom page feature.
+
+This is especially interesting to create geospatial widgets, such as a histogram or a category-view where instead of writing all the code by hand every time you can use just a simple a custom element.
+
+
+```html
+<as-category-widget
+  heading="Business Volume"
+  description="Description"
+  default-bar-color="#47DB99">
+</as-category-widget>
+```
+
+### Attributes
+
+[Attributes](https://en.wikipedia.org/wiki/HTML_attribute) modify the default functionality of an element. Usually the attributes will have a default value and using them will be optional. In the example above, heading and description are required attributes while the bar color is optional. 
+
+HTML Attributes can be only of type `string`, so if you want to use an object as an attribute you need to use javascript.
+
+Although you can use the [setAttribute](https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttribute) method, you can directly assign the attributes as properties of the element.
```

Comment:
You can assign the attributes to the element properties, or use [setAttribute](https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttribute) method to do it in a different way.

Do you think it's better?